zerver/tests/test_users.py

```diff
@@ -961,6 +961,17 @@ def test_get_user_by_id_in_realm_including_cross_realm(self) -> None:
             get_user_by_id_in_realm_including_cross_realm(
                 hamlet.id, None)
 
+    def test_get_user_subscription_status(self) -> None:
+        self.login('hamlet')
+        iago = self.example_user('iago')
+        stream = get_stream('Rome', iago.realm)
+        result = ujson.loads(self.client_get("/json/users/{}/subscription/{}".format(iago.id, stream.id)).content)
+        self.assertFalse(result['user']['is_subscribed'])
+
+        self.subscribe(iago, stream.name)
+        result = ujson.loads(self.client_get("/json/users/{}/subscription/{}".format(iago.id, stream.id)).content)
+        self.assertTrue(result['user']['is_subscribed'])
```

Comment:
Can you add a test that's trying an invalid user and invalid stream?

Reply:
Also, can you use `queries_captured` to validate the number of database queries done by this request?  We'd like to keep that small and it's important enough to be worth testing.